pkg/prometheus/promcfg_test.go

```diff
@@ -6803,6 +6803,222 @@ func TestScrapeConfigSpecConfigWithDigitalOceanSD(t *testing.T) {
 	}
 }
 
+func TestScrapeConfigSpecConfigWithDockerSDConfig(t *testing.T) {
+	for _, tc := range []struct {
+		name   string
+		scSpec monitoringv1alpha1.ScrapeConfigSpec
+		golden string
+	}{
+		{
+			name: "docker_sd_config_with_filters",
+			scSpec: monitoringv1alpha1.ScrapeConfigSpec{
+				DockerSDConfigs: []monitoringv1alpha1.DockerSDConfig{
```

Comment:
Whoops, forgot to select the lines I was referring to :P

Reply:
OAuth2.0 and Authorization are supposed to be mutually exclusive. So I think to reduce it to 2 test cases, from 4. Would that be fine?

Reply:
Oh right! Hmmmm, I'm not sure... it doesn't hurt that much to keep the 4 test cases, I was just nitpicking

Reply:
I did feel like having a separate test case for host_name was not really necessary. So I brought it down to three, one for each of the mutually exclusive OAuth, Auth and BasicAuth fields.